src/Symfony/Component/Filesystem/Filesystem.php

```diff
@@ -649,6 +649,29 @@ public function dumpFile($filename, $content)
     }
 
     /**
+     * Appends content to an existing file.
+     *
+     * @param string $filename The file for which to append content
+     * @param string $content  The content to append
+     *
+     * @throws IOException If the file is not writable
+     */
+    public function appendToFile($filename, $content)
+    {
+        $dir = dirname($filename);
+
+        if (!is_dir($dir)) {
+            $this->mkdir($dir);
```

Comment:
Don't we have to check here too that the directory is writable after we created it (the directory might have been created by another process)?